pkg/controller/gardenletdeployer/actuator.go

```diff
@@ -451,71 +445,25 @@ func (a *Actuator) reconcileSeedSecrets(ctx context.Context, obj client.Object,
 		return nil
 	}
 
-	// If backup is specified and DoNotCopyBackupCredentials feature gate is disabled,
-	// create or update the backup secret if it doesn't exist or is owned by the object.
-	var (
-		checksum     string
-		allowCopying = !utilfeature.DefaultFeatureGate.Enabled(features.DoNotCopyBackupCredentials)
-	)
-
 	// Get backup secret
 	backupSecret, originalErr := kubernetesutils.GetSecretByObjectReference(ctx, a.GardenClient, spec.Backup.CredentialsRef)
 
 	if client.IgnoreNotFound(originalErr) != nil {
 		return originalErr
 	}
 
-	if apierrors.IsNotFound(originalErr) && !allowCopying {
-		return fmt.Errorf("the configured backup secret does not exist, however the feature gate DoNotCopyBackupCredentials is enabled and shoot infrastructure credentials will not be reused: %w", originalErr)
-	}
-
-	if originalErr == nil {
-		checksum = utils.ComputeSecretChecksum(backupSecret.Data)[:8]
+	if apierrors.IsNotFound(originalErr) {
+		return fmt.Errorf("the configured backup secret does not exist and shoot infrastructure credentials will not be reused: %w", originalErr)
 	}
 
 	const (
 		secretStatusLabelKey   = "secret.backup.gardener.cloud/status"
```

Comment:
What is the destiny of this label key? I guess we cannot have a logic that cleans up all Secrets having this label because the same Secret name might be reused as backup Secret. If the expectation is components deploying the gardener-operator to have a cleanup logic for these Secrets, then let's document this as a release note to the PR.